File: app/src/main/java/com/example/calculator/SettingsActivity.java
package com.example.calculator;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

public class SettingsActivity extends AppCompatActivity {

    public static final String KEY_BUNDLE = "BUNDLE_THEMES";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        Intent intent = new Intent();

        findViewById(R.id.first_themes).setOnClickListener(v -> {
            intent.putExtra(KEY_BUNDLE, ApplicationThemes.ONE_THEMES.getKey());
            setResult(Activity.RESULT_OK, intent);
            finish();
        });

        findViewById(R.id.second_themes).setOnClickListener(v -> {
            intent.putExtra(KEY_BUNDLE, ApplicationThemes.TWO_THEMES.getKey());
            setResult(Activity.RESULT_OK, intent);
            finish();
        });

        findViewById(R.id.third_themes).setOnClickListener(v -> {
            intent.putExtra(KEY_BUNDLE, ApplicationThemes.THREE_THEMES.getKey());
            setResult(Activity.RESULT_OK, intent);
            finish();
        });
    }
}

File: app/src/main/java/com/example/calculator/SettingsActivity.kt
package com.example.calculator

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.calculator.databinding.ActivitySettingsBinding

class SettingsActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySettingsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySettingsBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val intent = Intent()
        binding.apply {
            firstThemes.setOnClickListener {
                intent.putExtra(KEY_BUNDLE, ApplicationThemes.ONE_THEMES.key)
                setResult(RESULT_OK, intent)
                finish()
            }

            secondThemes.setOnClickListener {
                intent.putExtra(KEY_BUNDLE, ApplicationThemes.TWO_THEMES.key)
                setResult(RESULT_OK, intent)
                finish()
            }

            thirdThemes.setOnClickListener {
                intent.putExtra(KEY_BUNDLE, ApplicationThemes.THREE_THEMES.key)
                setResult(RESULT_OK, intent)
                finish()
            }
        }
    }

    companion object {
        const val KEY_BUNDLE = "BUNDLE_THEMES"
    }
}